### PR TITLE
[qmf] Correctly handle multipart/signed parts

### PR DESCRIPTION
**Pull request by VDVsx**
multipart/signed parts can be the top level part having any other valid
containers inside as per RFC1847.

### CODE REVIEW

File: qmf/src/libraries/qmfclient/qmailmessage.cpp [hunk 1]
```diff
@@ -1047,6 +1047,9 @@ namespace findBody
         case QMailMessagePart::MultipartNone:
             return inMultipartNone(part, ctx);
 
+        case QMailMessagePart::MultipartMixed:
+            return inMultipartMixed(part, ctx);
+
```
Comment (VDVsx) [inline qmf/src/libraries/qmfclient/qmailmessage.cpp]: This is the default already below in line 1061, but is behind a warning, so adding here as well to make is more clear.

**Comment by VDVsx** [comment]
@matthewvogt Please have a look. Thx.

File: qmf/src/libraries/qmfclient/qmailmessage.cpp [hunk 2]
```diff
@@ -1153,6 +1156,20 @@ namespace findAttachments
             if (container.multipartType() == QMailMessagePart::MultipartMixed)
                 inMultipartMixed(container, found, hasAttachments);
 
+            if (container.multipartType() == QMailMessagePart::MultipartSigned) {
```
Comment (matthewvogt) [inline qmf/src/libraries/qmfclient/qmailmessage.cpp]: Put this loop into a 'inMultipartSigned' function to be consistent with the existing code.

**Comment by matthewvogt** [comment]
Otherwise looks good.

**Comment by VDVsx** [comment]
Updated according the suggestion.

**Comment by matthewvogt** [comment]
LGTM

**Pull request merged**